3495/0/triangle.rb

```diff
@@ -0,0 +1,77 @@
+# rubocop:disable Metrics/CyclomaticComplexity, Metrics/PerceivedComplexity, Style/GuardClause
+
+# triangle Project Code.
+
+# triangle analyzes the lengths of the sides of a triangle
+# (represented by a, b and c) and returns the type of triangle.
+#
+# It returns:
+#   :equilateral  if all sides are equal
+#   :isosceles    if exactly 2 sides are equal
+#   :scalene      if no sides are equal
+#
+# The tests for this method can be found in
+#   about_triangle_project.rb
+# and
+#   about_triangle_project_2.rb
+#
+# :reek:UtilityFunction
+class Triangle
+  attr_reader :side_a, :side_b, :side_c
+
+  def initialize(side_a, side_b, side_c)
+    @side_a = side_a
+    @side_b = side_b
+    @side_c = side_c
+  end
+
+  def type
+    if a_triangle?(self) || a_triangle_2?(self) || a_triangle_3?(self)
+      raise TriangleError, 'the triangle does not exist'
+    elsif a_equilateral?(self)
+      :equilateral
+    elsif a_isosceles?(self)
+      :isosceles
+    elsif a_scalene?(self)
+      :scalene
+    end
+  end
+
+  def a_triangle?(triangle)
```

Comment:
Look. 

1) First of all you don't need to send `triangle` to this method. You putted it to initializer. So `side_a, side_b, side_c` is visible for whole class.

2) Why are you returning `true` here?  `(triangle.side_a + triangle.side_b <= triangle.side_c) || (triangle.side_a + triangle.side_c <= triangle.side_b)` is already `true` or `false`. You can return just it. Something like that:
```
def a_triangle?
  (side_a + side_b <= side_c) || (side_a + side_c <= side_b)
end
```